osmaxx/excerptexport/views.py

```diff
@@ -225,3 +225,36 @@ def _social_identification_description(user):
         )
     else:
         return "not identified by any social identity providers"
+
+
+class ExcerptManageListView(ListView):
+    model = Excerpt
+    context_object_name = 'excerpts'
+    template_name = 'excerptexport/excerpt_manage_list.html'
+
+    def get_queryset(self):
+        return super().get_queryset().filter(owner=self.request.user, is_public=False)
+manage_own_excerpts = ExcerptManageListView.as_view()
+
+
+class DeleteExcerptView(DeleteView):
+    model = Excerpt
+    context_object_name = 'excerpt'
+    template_name = 'excerptexport/excerpt_confirm_delete.html'
+
+    def get_success_url(self):
+        return reverse('excerptexport:manage_own_excerpts')
+
+    def delete(self, request, *args, **kwargs):
+        excerpt = self.get_object()
+        if self.request.user != excerpt.owner:
+            raise GenericViewError("User doesn't match the excerpt's owner.")
+        if excerpt.is_public:
+            raise GenericViewError("No self-defined public excerpts can be deleted.")
```

Comment:
Should we also check whether the excerpt is referenced by any extraction orders not owned by the user? Or should users be allowed to delete their own previously public excerpts that have been set private e.g. via the admin interface?